source/dotnet/Library/AdaptiveCards.Rendering.Wpf/RendererUtil.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace AdaptiveCards.Rendering.Wpf
+{
+    static class RendererUtil
+    {
+        public static void ApplyIsVisible(FrameworkElement uiElement, AdaptiveElement element)
+        {
+            if (!element.IsVisible)
+            {
+                uiElement.Visibility = Visibility.Collapsed;
+            }
+        }
+
+        public static void ApplyVerticalContentAlignment(FrameworkElement uiElement, AdaptiveCollectionElement element)
+        {
+            switch (element.VerticalContentAlignment)
+            {
+                case AdaptiveVerticalContentAlignment.Center:
+                    uiElement.VerticalAlignment = VerticalAlignment.Center;
+                    break;
+                case AdaptiveVerticalContentAlignment.Bottom:
+                    uiElement.VerticalAlignment = VerticalAlignment.Bottom;
+                    break;
+                case AdaptiveVerticalContentAlignment.Top:
+                default:
+                    break;
+            }
+        }
+
+        public static FrameworkElement ApplySelectAction(FrameworkElement uiElement, AdaptiveElement element, AdaptiveRenderContext context)
+        {
+            if (element is AdaptiveCollectionElement)
+            {
+                AdaptiveCollectionElement collectionElement = element as AdaptiveCollectionElement;
+                if (collectionElement.SelectAction != null)
+                {
+                    return context.RenderSelectAction(collectionElement.SelectAction, uiElement);
+                }
+            }
+            else if (element is AdaptiveImage)
+            {
+                AdaptiveImage image = element as AdaptiveImage;
+                if (image.SelectAction != null)
+                {
+                    return context.RenderSelectAction(image.SelectAction, uiElement);
+                }
```

Comment:
You could simplify this a bit by pulling the null check and the call to context.RenderSelectAction out of the if. Something like this:
if (element is AdaptiveCollectionElement)
{
    selectAction = (element as AdaptiveCollectionElement).SelectAction
}
else if (element is AdaptiveImage)
{
    selectAction = (element as AdaptiveImage).SelectAction
}

if (selectAction != null)
{
    return context.RenderSelectAction(selectAction, uiElement);
}